File: content/courses/parallel-computing-introduction/codes/omp_hello.f90
program omp_hello
use omp_lib
implicit none

   integer :: i
   integer :: nthreads, tid

   !$omp parallel private(tid)
      nthreads=omp_get_num_threads()
      tid=omp_get_thread_num()
      write(*,'(a,i3,a,i3)') "Hello from thread ",tid," of ",nthreads
   !$omp end parallel

end program

File: content/courses/parallel-computing-introduction/codes/omp_hello.cxx
#include <iostream>
#include <cstdio>
#include <omp.h>

using namespace std;

int main() {

    int tid, nthreads;

    #pragma omp parallel private(tid)
    {
    nthreads=omp_get_num_threads();
    tid=omp_get_thread_num();
    printf("Hello from thread %d of %d\n",tid, nthreads);
    //cout<<"Hello from thread"<<" "<<tid<<" of"<<nthreads<<"\n";
    }
}
